src/main/scala/rocket/RocketCore.scala

```diff
@@ -875,8 +875,8 @@ class Rocket(tile: RocketTile)(implicit p: Parameters) extends CoreModule()(p)
   coreMonitorBundle.timer := csr.io.time(31,0)
   coreMonitorBundle.valid := csr.io.trace(0).valid && !csr.io.trace(0).exception
   coreMonitorBundle.pc := csr.io.trace(0).iaddr(vaddrBitsExtended-1, 0).sextTo(xLen)
-  coreMonitorBundle.wrdst := rf_waddr
-  coreMonitorBundle.wren := rf_wen && !(wb_set_sboard && wb_wen)
+  coreMonitorBundle.wrdst := Mux(wb_ctrl.wxd || wb_ctrl.wfd, wb_waddr, 0.U)
```

Comment:
I don't think this mux does anything useful, since `wrdst` is a don't-care when `wren` is false, and with your change, `wren` is only true when `wb_ctrl.wxd` is true.

Reply:
it's actually the inverse that we want to use to identify long-latency writebacks: when `wb_ctrl.wxd` is true and `wren` is false

Reply:
I don't see how that works, though: `wb_ctrl.wxd` and `wb_ctrl.wfd` are themselves don't-cares some of the time, namely, on cycles when there's nothing in the writeback stage and no long-latency writeback happening.

Furthermore, 0 doesn't work as a sentinel value when FP registers are involved, because f0 isn't hardwired.

It seems like you're trying to multiplex an additional control signal on top of this field... why not just add the control signal?

Reply:
I agree with your principles, and I'll have to think on it / discuss further for specifics.